Sources/Plasma/PubUtilLib/plClientResMgr/plClientResMgr.cpp

```diff
@@ -87,7 +87,10 @@ void plClientResMgr::ILoadResources(const plFileName& resfile)
                 for (int i = 0; i < num_resources; i++) {
                     plMipmap* res_data = nullptr;
                     uint32_t res_size = 0;
-                    ST::string res_name = in.ReadSafeStringLong();
+                    ST::char_buffer res_name;
+                    uint32_t numChars = in.ReadLE32();
+                    res_name.allocate(numChars);
+                    in.Read(numChars, res_name.data());
```

Comment:
This doesn't account for the Safe String bit flipping...

Reply:
That's on purpose. AFAICT, the resource.dat names are always stored in plain text and never use the bit-flipping obfuscation.

Reply:
I believe that's only true for ones we produce.  If you check the one used in Myst V, it has obfuscated strings.

Reply:
Hmm, I checked the resource.dat of every MOULa shard I have installed (Cyan, the Minkatas, Gehn, TOC, TL), and the file names are all plain strings, even the shards that ship non-H'uru clients. In my copy of Myst V (from GOG), the Cursors.dat also has plain file names. That's why I assumed that they were never obfuscated at any point.

(Besides, since this is Plasma and not libHSPlasma, is compatibility with pre-MOULa data files even important?)

Reply:
Oh, I must have misremembered that about Myst V.  I could have sworn it did...